### PR TITLE
UWP MESA 4k fix

### PR DESCRIPTION
**Pull request by aerisarn**
## Description

Fix for detecting resolutions higher than 1080p on XBOX UWP and Mesa. UWP API calls in MESA were misaligned
and reporting 1080p due to a known XBOX peculiarity. Besides, reading width and heigth must be done on UI thread so it might trigger unwanted waits, so caching values have been added to prevent multiple waits.

## Related Issues

No official issues, this is XBOX specific

## Related Pull Requests

No one, MESA was already integrated

## Reviewers

@GABO1423

### CODE REVIEW

**Comment by LibretroAdmin** [comment]
Any reason why the OPENGL ifdefs are gone? If it does not hurt, could you reintroduce them again or was there a reason for its removal?

File: pkg/msvc-uwp/RetroArch-msvcUWP/MESA/README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # MESA PreBuilt Libraries
 
 The Binaries in this folder are compiled from a custom version of [MESA](https://github.com/aerisarn/mesa-uwp). 
-These are based on this [tag](https://github.com/aerisarn/mesa-uwp/releases/tag/alpha-2-hack).
+These are based on this [branch](https://github.com/aerisarn/mesa-uwp/releases/tag/alpha-2-hack-fixes), which is the
```
Comment (CasperMcFadden95) [inline pkg/msvc-uwp/RetroArch-msvcUWP/MESA/README.md:4]: Is the URL correct?

Comment (aerisarn) [reply pkg/msvc-uwp/RetroArch-msvcUWP/MESA/README.md]: fixed

**Comment by aerisarn** [comment]
> ## Description
> Fix for detecting resolutions higher than 1080p on XBOX UWP and Mesa. UWP API calls in MESA were misaligned and reporting 1080p due to a known XBOX peculiarity. Besides, reading width and heigth must be done on UI thread so it might trigger unwanted waits, so caching values have been added to prevent multiple waits.
> 
> ## Related Issues
> No official issues, this is XBOX specific
> 
> ## Related Pull Requests
> No one, MESA was already integrated
> 
> ## Reviewers
> @GABO1423



> Any reason why the OPENGL ifdefs are gone? If it does not hurt, could you reintroduce them again or was there a reason for its removal?

They were removed because MESA was using another method to read those values which was not in synch with retroarch's one. HAVE_OPENGL was introduced to force that behavior when building against mesa (as there can be no OpenGL on UWP without it). Now that both MESA and Retroarch use the same way, there's no need to differentiate such behavior

**Comment by LibretroAdmin** [comment]
Alright then.

**Pull request merged**